### PR TITLE
fix(changelog/gitlab): custom endpoint and repository length validation

### PR DESCRIPTION
**Pull request by setchy**
## Changes

Fixes GitLab repository length validation and custom registry support

## Context

https://github.com/renovatebot/renovate/discussions/23156

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/update/pr/changelog/gitlab/source.ts [hunk 1]
```diff
@@ -18,4 +18,8 @@ export class GitLabChangeLogSource extends ChangeLogSource {
   ): string {
     return `${baseUrl}${repository}/compare/${prevHead}...${nextHead}`;
   }
+
+  override hasValidRepository(repository: string): boolean {
+    return repository.split('/').length >= 2;
```
Comment (viceice) [inline lib/workers/repository/update/pr/changelog/gitlab/source.ts:23]: is this the fastet way to check this? 🙃

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/workers/repository/update/pr/changelog/gitlab/source.ts (hunk 1 above)
Comment (rarkins) [reply lib/workers/repository/update/pr/changelog/gitlab/source.ts:23]: I think it's fast enough

File: lib/workers/repository/update/pr/changelog/source.ts [hunk 2]
```diff
@@ -42,6 +42,7 @@ export abstract class ChangeLogSource {
   async getAllTags(endpoint: string, repository: string): Promise<string[]> {
     const tags = (
       await getPkgReleases({
+        registryUrls: [endpoint],
```
Comment (viceice) [inline lib/workers/repository/update/pr/changelog/source.ts:45]: Can you create a test to validate it's passed correctly?

**Review by rarkins — APPROVED** [review]

File: lib/workers/repository/update/pr/changelog/gitlab/index.spec.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { partial } from '../../../../../../../test/util';
 import * as semverVersioning from '../../../../../../modules/versioning/semver';
 import * as hostRules from '../../../../../../util/host-rules';
 import type { BranchUpgradeConfig } from '../../../../../types';
+import { GitLabChangeLogSource } from './source';
```
Comment (HonkingGoose) [inline lib/workers/repository/update/pr/changelog/gitlab/index.spec.ts:7]: This PR targets `main` so will result in a new version of `v36` when merged. Is this fix critical enough to backport to the old `v35` release?

I found these "rules" in our docs: [^1]

> ## Which Renovate versions are officially supported?
> 
> Only the latest version of Renovate is supported by the Renovate maintainers.
> The Renovate team only fixes bugs in an older version if:
> 
> - the hosted app needs to stay on that older major version for a short time, or
> - some critical bug needs to be fixed and the new major is blocked
> 
> If you're using the hosted app, you don't need to do anything, as the Renovate maintainers update the hosted app regularly.
> If you're self hosting Renovate, use the latest release if possible.

[^1]: [Renovate docs, which Renovate versions are officially supported?](https://docs.renovatebot.com/faq/#which-renovate-versions-are-officially-supported)

Comment (viceice) [reply lib/workers/repository/update/pr/changelog/gitlab/index.spec.ts:7]: @rarkins WDYT? v35 publishing is already active, so backport should be easy

**Review by viceice — APPROVED** [review]

File: lib/workers/repository/update/pr/changelog/gitlab/index.spec.ts (hunk 3 above)
Comment (rarkins) [reply lib/workers/repository/update/pr/changelog/gitlab/index.spec.ts:7]: Yes, let's backport

**Pull request merged**